official/resnet/resnet_run_loop.py

```diff
@@ -260,19 +280,21 @@ def exclude_batch_norm(name):
 
     optimizer = tf.train.MomentumOptimizer(
         learning_rate=learning_rate,
-        momentum=momentum)
-
-    # If we are running multi-GPU, we need to wrap the optimizer.
-    if multi_gpu:
-      optimizer = tf.contrib.estimator.TowerOptimizer(optimizer)
+        momentum=momentum
+    )
 
     update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
     train_op = tf.group(optimizer.minimize(loss, global_step), update_ops)
   else:
     train_op = None
 
-  accuracy = tf.metrics.accuracy(
-      tf.argmax(labels, axis=1), predictions['classes'])
+  if not distribute_lib.has_distribution_strategy():
```

Comment:
this can be `tf.contrib.distribute.has_distribution_strategy()`

Reply:
If we can check in code whether we have a dist strat via a tf function, why do we need to pass around use_distribution_strategies?

Reply:
You're absolutely right @karmel. Anytime one is in the distribution strategy scope, the `has_distribution_strategy` check should work. I believe in our earlier code, we didn't have the input function running inside the scope, so we had to pass around this boolean. But Since then we've moved input processing to be under distribution scope as well and can now use `has_distribution_strategy()` almost everywhere. 

I just checked where all it's used. I think it can be replaced with `has_distribution_strategy` in all places except this one in `resnet_main` . And you already suggested removing that check from there entirely. So I think we can get rid of this flag entirely!